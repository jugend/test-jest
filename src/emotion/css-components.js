import React from 'react'
import { css, cx } from 'emotion'

const blueStyle = css`
  color: blue;
`

const styles = {
  bordered: css`
    border: 1px solid blue;
  `
}

export function BlueButton(props) {
  return (
    <button className={cx(blueStyle, styles.bordered)} data-testid="blue-button">
      {props.children}
    </button>
  )
}
