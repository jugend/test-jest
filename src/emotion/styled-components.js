import React from 'react'
import styled, { css } from 'react-emotion'

export const BlueButton = styled('button')`
  color: blue;
`

export const BorderedButton = styled.button((props) => `
  border: 1px solid ${props.borderColor}
`)

// export const BorderedButton = styled.button((props) => ({
//   border: `1px solid ${props.borderColor}`
// }))
