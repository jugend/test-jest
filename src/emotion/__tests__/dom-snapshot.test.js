import React from 'react'
import renderer from 'react-test-renderer'
import * as emotion from 'emotion'

describe('jest snapshot printer', () => {
  const divStyle = emotion.css(`
    color: red;
  `)

  const inputStyle = emotion.css(`
    background-color: blue;
  `)

  it('prints React elements', () => {
    expect(
      renderer
        .create(
          <div className={divStyle}>
            <input className={inputStyle} />
          </div>
        )
        .toJSON()
    ).toMatchSnapshot()
  })

  it('prints DOM elements', () => {
    const divElement = document.createElement('div')
    divElement.setAttribute('class', divStyle)
    const inputElement = document.createElement('input')
    inputElement.setAttribute('class', inputStyle)
    divElement.appendChild(inputElement)

    expect(divElement).toMatchSnapshot()
  })
})
