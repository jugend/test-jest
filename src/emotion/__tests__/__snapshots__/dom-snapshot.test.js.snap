// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshot printer prints DOM elements 1`] = `
.emotion-1 {
  color: red;
}

.emotion-0 {
  background-color: blue;
}

<div
  class="emotion-1"
>
  <input
    class="emotion-0"
  />
</div>
`;

exports[`jest snapshot printer prints React elements 1`] = `
.emotion-1 {
  color: red;
}

.emotion-0 {
  background-color: blue;
}

<div
  className="emotion-1"
>
  <input
    className="emotion-0"
  />
</div>
`;
